src/conditional.jl

```diff
@@ -1,59 +1,67 @@
+
+## Links:
+##     Thesis - http://www.cs.nyu.edu/~gwtaylor/thesis/Taylor_Graham_W_200911_PhD_thesis.pdf
+##     FCRBM - http://www.cs.toronto.edu/~fritz/absps/fcrbm_icml.pdf
+
+
+# Input data layout for 2 batches and 3 steps of history
+#
+#      batch 1               batch 2
+# |--------------------|--------------------|-----
+# |  current visible   |  current visible   | ...
+# |--------------------|--------------------|-----
+# |  history step 1    |  history step 1    | ...
+# |  history step 2    |  history step 2    | ...
+# |  history step 3    |  history step 3    | ...
+# |--------------------|--------------------|-----
+
+
 import StatsBase: predict
 
 
-@runonce type ConditionalRBM{V,H} <: AbstractRBM{V,H}
-    W::Matrix{Float64}  # standard weights
-    A::Matrix{Float64}  # autoregressive params (vis to vis)
-    B::Matrix{Float64}  # hidden params(vis to hid)
-    dyn_vbias::Array{Float64}
-    dyn_hbias::Array{Float64}
-    vbias::Vector{Float64}
-    hbias::Vector{Float64}
+@runonce type ConditionalRBM{T,V,H} <: AbstractRBM{T,V,H}
+    W::Matrix{T}  # standard weights
+    A::Matrix{T}  # autoregressive params (vis to vis)
+    B::Matrix{T}  # hidden params(vis to hid)
+    vbias::Vector{T}
+    hbias::Vector{T}
+    dyn_vbias::Array{T}
+    dyn_hbias::Array{T}
     steps::Int
-    dW_prev::Matrix{Float64}
-    persistent_chain::Matrix{Float64}
-    momentum::Float64
-    decay_rate::Float64
-    sparsity_cost::Float64
-    sparsity_exp::Float64
+end
+
+
+function ConditionalRBM(T::Type, V::Type, H::Type,
+                        n_vis::Int, n_hid::Int, steps::Int; sigma=0.01)
+    ConditionalRBM{T,V,H}(
+        map(T, rand(Normal(0, sigma), (n_hid, n_vis))),
+        map(T, rand(Normal(0, sigma), (n_vis, n_vis * steps))),
+        map(T, rand(Normal(0, sigma), (n_hid, n_vis * steps))),
+        zeros(T, n_vis),
+        zeros(T, n_hid),
+        zeros(T, n_vis),
+        zeros(T, n_hid),
+        steps)
 end
 
 function ConditionalRBM(V::Type, H::Type, n_vis::Int, n_hid::Int;
-        sigma=0.001, momentum=0.9, decay_rate=0.0, sparsity_cost=0.0,
-        sparsity_exp=0.5, steps=5)
-
-    ConditionalRBM{V,H}(
-        rand(Normal(0, sigma), (n_hid, n_vis)),
-        rand(Normal(0, sigma), (n_vis, n_vis * steps)),
-        rand(Normal(0, sigma), (n_hid, n_vis * steps)),
-        zeros(n_vis),
-        zeros(n_hid),
-        zeros(n_vis),
-        zeros(n_hid),
-        steps,
-        zeros(n_hid, n_vis),
-        Array(Float64, 0, 0),
-        momentum,
-        decay_rate,
-        sparsity_cost,
-        sparsity_exp
-    )
-end
-
-function Base.show{V,H}(io::IO, rbm::RBM{V,H})
-    n_vis = size(rbm.vbias, 1)
-    n_hid = size(rbm.hbias, 1)
-    print(io, "RBM{$V,$H}($n_vis, $n_hid)")
-end
-
-function split_vis(crbm::ConditionalRBM, vis::Mat{Float64})
+                        steps=5, sigma=0.01)
+    ConditionalRBM(Float64, Bernoulli, Bernoulli, n_vis, n_hid, steps;
```

Comment:
I think you want this to be `ConditionalRBM(Float64, V, H, n_vis, n_hid, ...`? If so I can change that when I add my generalize changes.